src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -720,6 +720,14 @@ public function getSession()
         $session = $this->session;
         if (!$session instanceof SessionInterface && null !== $session) {
             $this->setSession($session = $session());
+            /*
+             * For supporting sessions in php runtime with runners like roadrunner or swoole the session
+             * cookie need read from the cookie bag and set on the session storage.
+             */
+            if (!$session->isStarted()) {
+                $sessionId = $this->cookies->get($session->getName(), '');
+                $session->setId($sessionId);
```

Comment:
@jderusse It seems like this can not happend, it can only regenerate a session when its already started else PHP will throw a warning:

```bash
php -r 'session_regenerate_id();'
```
Will output:

```
PHP Warning:  session_regenerate_id(): Session ID cannot be regenerated when there is no active session in Command line code on line 1

Warning: session_regenerate_id(): Session ID cannot be regenerated when there is no active session in Command line code on line 1
```

Reply:
I could also check if there is no sessionId already set? What do you think about this?

Reply:
Method was like suggested above moved to the AbstractSessionListener which I think is great so all logic about this lives in the HttpKernel component now.